pkg/cloudprovider/providers/openstack/openstack.go

```diff
@@ -1123,3 +1123,23 @@ func (os *OpenStack) DeleteVolume(volumeName string) error {
 	}
 	return err
 }
+
+// Get device path of attached volume to the compute running kubelet
+func (os *OpenStack) GetAttachmentDiskPath(instanceID string, diskName string) (string, error) {
+	disk, err := os.getVolume(diskName)
+	if err != nil {
+		return "", err
+	}
+	if len(disk.Attachments) > 0 && disk.Attachments[0]["server_id"] != nil {
+		if instanceID == disk.Attachments[0]["server_id"] {
+			// Attachment[0]["device"] points to the device path
+			// see http://developer.openstack.org/api-ref-blockstorage-v1.html
+			return disk.Attachments[0]["device"].(string), nil
```

Comment:
Is this reliable? In our OS1 I can see that my volume should be attached as /dev/vdc, but in the VM I can see it as /dev/vdb. IMO the device in OSP API is just "recommended" name, not the actual name.